### ISSUE TITLE
Compilation and linking of simdis examples on x86-64 linux platform

### ISSUE THREAD

**Issue by Bj29ASH** [USNavalResearchLaboratory/simdissdk#115]
I am trying to compile the basic viewer example on ubuntu-20 intel x86 platform. Facing link time errors.

--------------------------------------------------------------------------------------------------------------------------------------------

futura@futura:~/workspace/simdissdk-simdissdk-1.16/Examples/BasicViewer/build$ cmake ..
-- The C compiler identification is GNU 9.4.0
-- The CXX compiler identification is GNU 9.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Warning (dev) in CMakeLists.txt:
  No cmake_minimum_required command is present.  A line of code such as

    cmake_minimum_required(VERSION 3.16)

  should be added at the top of the file.  The version specified may be lower
  if you wish to support older CMake versions for this project.  For more
  information run "cmake --help-policy CMP0000".
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Configuring done
-- Generating done
-- Build files have been written to: /home/futura/workspace/simdissdk-simdissdk-1.16/Examples/BasicViewer/build
futura@futura:~/workspace/simdissdk-simdissdk-1.16/Examples/BasicViewer/build$ make
Scanning dependencies of target example_basicviewer
[ 50%] Building CXX object CMakeFiles/example_basicviewer.dir/BasicViewer.o
[100%] Linking CXX executable example_basicviewer
/usr/bin/ld: CMakeFiles/example_basicviewer.dir/BasicViewer.o: undefined reference to symbol '_ZN7simCore9soVersionEv'
/usr/bin/ld: /home/futura/workspace/SIMDIS_SDK_BIN_linux64_gcc-8.3/SIMDIS_SDK/lib/libsimCore.so.16: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/example_basicviewer.dir/build.make:86: example_basicviewer] Error 1
make[1]: *** [CMakeFiles/Makefile2:76: CMakeFiles/example_basicviewer.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
futura@futura:~/workspace/simdissdk-simdissdk-1.16/Examples/BasicViewer/build$
--------------------------------------------------------------------------------------------------------------------------------------

Here is the CMakeLists.txt file of the basic viewer -
--------------------------------------------------------------------------------------------------------------------------------------
project(EXAMPLE_BASIC_VIEWER)

set(PROJECT_FILES
    BasicViewer.cpp
)

add_executable(example_basicviewer ${PROJECT_FILES})
target_include_directories(example_basicviewer PRIVATE
    /home/futura/workspace/SIMDIS_SDK_BIN_linux64_gcc-8.3/SIMDIS_SDK/include
    "/home/futura/workspace/SIMDIS_SDK_3rd_linux64_gcc-8.3/SIMDIS_SDK/3rd/linux64_gcc-8.3/OpenSceneGraph/3.6.5/include"
    "/home/futura/workspace/SIMDIS_SDK_3rd_linux64_gcc-8.3/SIMDIS_SDK/3rd/linux64_gcc-8.3/osgEarth/OSG-3.6.5/include"
    "/home/futura/workspace/SIMDIS_SDK_BIN_linux64_gcc-8.3/SIMDIS_SDK/include"
    "/home/futura/workspace/mpds-official/mpds_linux_3rd_party/geographic-lib-1.52/include"
    "/home/futura/workspace/SIMDIS_SDK_3rd_linux64_gcc-8.3/SIMDIS_SDK/3rd/linux64_gcc-8.3/osgQt/3.6.5/include"
    "/home/futura/workspace/SIMDIS_SDK_3rd_linux64_gcc-8.3/SIMDIS_SDK/3rd/linux64_gcc-8.3/protobuf/3.14.0/include"
    "/home/futura/workspace/mpds-official/mpds_linux_3rd_party/imgui-1.83/include"
    "/home/futura/workspace/mpds-official/mpds_linux_3rd_party/sqlite-modern-cpp-3.2.8/include"
    "/home/futura/workspace/SIMDIS_SDK_3rd_linux64_gcc-8.3/SIMDIS_SDK/3rd/linux64_gcc-8.3/SQLite/3.36.0/include"
    "/home/futura/workspace/SIMDIS_SDK_3rd_linux64_gcc-8.3/SIMDIS_SDK/3rd/linux64_gcc-8.3/GLEW/2.1.0/include"
    "/home/futura/workspace/SIMDIS_SDK_3rd_linux64_gcc-8.3/SIMDIS_SDK/3rd/linux64_gcc-8.3/GDAL/3.2.3/include"
    "/home/futura/workspace/mpds-official/mpds_linux_3rd_party/polylabel-1.1.0/include"
    "/home/futura/workspace/mpds-official/mpds_linux_3rd_party/variant-1.1.3/include"
    "/home/futura/workspace/mpds-official/mpds_linux_3rd_party/geometry.hpp-2.0.3/include"
)
#target_link_directories(example_basicviewer PRIVATE "/home/futura/workspace/SIMDIS_SDK_BIN_linux64_gcc-8.3/SIMDIS_SDK/lib")

target_link_libraries(example_basicviewer PRIVATE
    /home/futura/workspace/SIMDIS_SDK_BIN_linux64_gcc-8.3/SIMDIS_SDK/lib/libsimVis.so.1.16.0
    /home/futura/workspace/SIMDIS_SDK_BIN_linux64_gcc-8.3/SIMDIS_SDK/lib/libsimUtil.so.1.16.0
)

set_target_properties(example_basicviewer PROPERTIES
    FOLDER "Examples"
    PROJECT_LABEL "Basic Viewer"
)

if (TARGET ImGuiLib)
    target_link_libraries(example_basicviewer PRIVATE ImGuiLib)
    target_compile_definitions(example_basicviewer PRIVATE HAVE_IMGUI)
endif()

#vsi_install_target(example_basicviewer SDK_Examples)
--------------------------------------------------------------------------------------------------------------------------------------

Could you please provide the appropriate cmake command to compile and link the basic viewer example.
Basically how to build the examples on linux?

**Comment by emminizer** [comment]
basic_viewer is not standalone and is part of the larger SIMDIS SDK CMake project. The top level CMake file is a few directories up. See https://github.com/USNavalResearchLaboratory/simdissdk/blob/main/CMakeLists.txt
